bigframes/core/compile/compiled.py

```diff
@@ -45,76 +42,250 @@
 PREDICATE_COLUMN = "bigframes_predicate"
 
 
-class CompiledArrayValue:
-    """Immutable BigQuery DataFrames expression tree.
+class CompiledArrayValue(typing.Protocol):
+    @property
+    def column_ids(self) -> typing.Sequence[str]:
+        ...
 
-    Note: Usage of this class is considered to be private and subject to change
-    at any time.
+    def to_sql(self) -> str:
```

Comment:
Does this need to take more parameters?

Reply:
No, if the caller called in unordered mode, it also know that it has no parameters to pass here - all parameters relate to ordering.